src/signaling/backend_configuration.go

```diff
@@ -148,6 +148,101 @@ func NewBackendConfiguration(config *goconf.ConfigFile) (*BackendConfiguration,
 	}, nil
 }
 
+func (b *BackendConfiguration) RemoveBackend(host string) {
+	delete(b.backends, host)
+}
+
+func (b *BackendConfiguration) UpsertHost(host string, backends []*Backend) {
+	existingIndex := 0
+	for _, existingBackend := range b.backends[host] {
+		found := false
+		index := 0
+		for _, newBackend := range backends {
+			if existingBackend == newBackend {
+				found = true
+				backends = append(backends[:index], backends[index+1:]...)
+				continue
```

Comment:
Shouldn't this be a `break`? You found the backend, so no need to further iterate the list.

Also: I don't see how a secret of an existing backend would be changed.